web-app/js/portal/cart/BaseInjector.js

```diff
@@ -66,6 +70,7 @@ Portal.cart.BaseInjector = Ext.extend(Object, {
 
     _addDownloadEstimate: function(dataCollection) {
 
+        dataCollection.uuid = this._getCollectionUuid(dataCollection);
```

Comment:
I don't think it's a good idea to mutate the `dataCollection` object here.

`dataCollection.uuid` should be set when the `dataCollection` object is first instantiated.